disco/disco.go

```diff
@@ -277,10 +294,12 @@ func (d *Disco) discover(hostname svchost.Hostname) (*Host, error) {
 // Forget invalidates any cached record of the given hostname. If the host
 // has no cache entry then this is a no-op.
 func (d *Disco) Forget(hostname svchost.Hostname) {
+	delete(d.aliasMap, hostname)
```

Comment:
I don't think I'd expect `Forget` to forget an alias, based on the documentation of this function.

It sounds like this was intended to be for purging the cache of previously-discovered hostnames, and so if there are any existing callers of this they're probably expecting a subsequent request for the same hostname to still return an equivalent result but to re-check it against the real server.

Reply:
Good catch. I added a ForgetAlias method for completeness.